rotkehlchen/data/importer.py

```diff
@@ -1275,3 +1323,97 @@ def import_uphold_transactions_csv(self, filepath: Path, **kwargs: Any) -> Tuple
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    def _consume_bisq_trade(self, csv_row: Dict[str, Any], **kwargs: Any) -> None:
+        """
+        Consume the file containing only trades from Bisq.
+        - UnknownAsset
+        - DeserializationError
+        """
+        if csv_row['Status'] == 'Canceled':
+            return
+        formatstr = kwargs.get('timestamp_format')
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date/Time'],
+            formatstr=formatstr if formatstr is not None else '%d %b %Y %H:%M:%S',
+            location='Bisq',
+        )
+        # Get assets and amounts sold
+        offer = csv_row['Offer type'].split()
+        assets1_symbol, assets2_symbol = csv_row['Market'].split('/')
+        if offer[0] == 'Sell':
+            trade_type = TradeType.SELL
+            if offer[1] == assets1_symbol:
+                base_asset = symbol_to_asset_or_token(assets1_symbol)
+                quote_asset = symbol_to_asset_or_token(assets2_symbol)
+            else:
+                base_asset = symbol_to_asset_or_token(assets2_symbol)
+                quote_asset = symbol_to_asset_or_token(assets1_symbol)
+
+            if base_asset == A_BTC:
+                buy_amount = deserialize_asset_amount(csv_row['Amount'])
+            else:
+                buy_amount = deserialize_asset_amount(csv_row['Amount in BTC'])
+        else:
+            trade_type = TradeType.BUY
+            if offer[1] == assets1_symbol:
+                base_asset = symbol_to_asset_or_token(assets1_symbol)
+                quote_asset = symbol_to_asset_or_token(assets2_symbol)
+            else:
+                base_asset = symbol_to_asset_or_token(assets2_symbol)
+                quote_asset = symbol_to_asset_or_token(assets1_symbol)
+
+            if base_asset == A_BTC:
+                buy_amount = deserialize_asset_amount(csv_row['Amount in BTC'])
+            else:
+                buy_amount = deserialize_asset_amount(csv_row['Amount'])
```

Comment:
I no longer see the comment I had about this code appear here. So I am gonna rewrite it.

I saw a lot of repetition here so I thought this could be improved. But writing it out it seems not easy:

```python
trade_type = TradeType.SELL if offer[0] == 'Sell' else TradeType.BUY
if offer[1] == assets1_symbol:
    base_asset = symbol_to_asset_or_token(assets1_symbol)
    quote_asset = symbol_to_asset_or_token(assets2_symbol)
else:
    base_asset = symbol_to_asset_or_token(assets2_symbol)
    quote_asset = symbol_to_asset_or_token(assets1_symbol)

amount_key = 'Amount' if (base_asset == A_BTC and trade_type == TradeType.SELL) or (base_asset != A_BTC and trade_type == TradeType.BUY) else 'Amount'  # noqa: E501
buy_amount = deserialize_asset_amount(csv_row[amount_key])
```

Closest is this, which is much shorter and nice, but don't like the extra huge condition on amount key